### PR TITLE
Integrate geospatial ip2geo functionality with security plugin

### PR DESCRIPTION
**Pull request by heemin32**
### Description
Adds 2 ip2geo related roles to the plugin: read only and fully access.
Adds ip2geo system index to demo scripts as well.

### Issues Resolved
N/A

### Testing
N/A

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tools/install_demo_configuration.bat [hunk 1]
```diff
@@ -315,7 +315,7 @@ echo plugins.security.enable_snapshot_restore_privilege: true >> "%OPENSEARCH_CO
...
+echo plugins.security.system_indices.indices: [".plugins-ml-config", ".plugins-ml-connector", ".plugins-ml-model-group", ".plugins-ml-model", ".plugins-ml-task", ".opendistro-alerting-config", ".opendistro-alerting-alert*", ".opendistro-anomaly-results*", ".opendistro-anomaly-detector*", ".opendistro-anomaly-checkpoints", ".opendistro-anomaly-detection-state", ".opendistro-reports-*", ".opensearch-notifications-*", ".opensearch-notebooks", ".opensearch-observability", ".ql-datasources", ".opendistro-asynchronous-search-response*", ".replication-metadata-store", ".opensearch-knn-models", ".geospatial.ip2geo.data.*"] >> "%OPENSEARCH_CONF_FILE%"
```
Comment (cwperks) [inline tools/install_demo_configuration.bat:318]: Curious, is there any reason why you are using `.` as a delimiter in the system index name? In all other cases in this list, plugins use hyphens. While its possible to use dot separation, would hyphen be better for consistency?

**Comment by cwperks** [comment]
@heemin32 Is this change targeting 2.10?

**Comment by heemin32** [comment]
> @heemin32 Is this change targeting 2.10?

Yes

File: tools/install_demo_configuration.bat (hunk 1 above)
Comment (heemin32) [reply tools/install_demo_configuration.bat:318]: Make sense. Let me update it.

**Review by cwperks — APPROVED** [review]

**Comment by cwperks** [comment]
@heemin32 Can you rebase this PR with main? There were some recent build fixing PRs merged.

**Comment by heemin32** [comment]
> @heemin32 Can you rebase this PR with main? There were some recent build fixing PRs merged.

There is no conflict. So, rebase is not needed.

**Comment by heemin32** [comment]
> @heemin32 Can you rebase this PR with main? There were some recent build fixing PRs merged.

Thanks! Rebased.

**Comment by cwperks** [comment]
Can another maintainer please review this PR?

**Review by peternied — APPROVED** [review]

**Pull request merged**